lighthouse-core/computed/metrics/cumulative-layout-shift-all-frames.js

```diff
@@ -5,30 +5,70 @@
  */
 'use strict';
 
+/** @typedef {Omit<LH.TraceEvent, 'name'|'args'> & {name: 'LayoutShift', args: {data: {score: number, weighted_score_delta: number, had_recent_input: boolean}}}} LayoutShiftEvent */
+
 const makeComputedArtifact = require('../computed-artifact.js');
 const TraceOfTab = require('../trace-of-tab.js');
+const log = require('lighthouse-logger');
 
 class CumulativeLayoutShiftAllFrames {
+  /**
+   * @param {LH.TraceEvent} event
+   * @return {event is LayoutShiftEvent}
+   */
+  static isLayoutShiftEvent(event) {
+    return Boolean(
+      event.name === 'LayoutShift' &&
+      event.args &&
+      event.args.data &&
+      event.args.data.score !== undefined
+    );
+  }
+
   /**
    * @param {LH.Trace} trace
    * @param {LH.Audit.Context} context
    * @return {Promise<{value: number}>}
    */
   static async compute_(trace, context) {
     const traceOfTab = await TraceOfTab.request(trace, context);
-    const cumulativeShift = traceOfTab.frameTreeEvents
-      .filter(e =>
-        e.name === 'LayoutShift' &&
-        e.args &&
-        e.args.data &&
-        e.args.data.score &&
-        !e.args.data.had_recent_input
-      )
+    const layoutShiftEvents = traceOfTab.frameTreeEvents.filter(this.isLayoutShiftEvent);
+
+    // Chromium will set `had_recent_input` if there was recent user input, which
+    // skips shift events from contributing to CLS. This flag is also set when Lighthouse changes
+    // the emulation size. This consistently results in the first few shift event always being
+    // ignored for CLS. Since we don't expect any user input, we add the score of these
+    // shift events to CLS.
+    // See https://bugs.chromium.org/p/chromium/issues/detail?id=1094974.
+    for (const event of layoutShiftEvents) {
+      if (!event.args.data.had_recent_input) break;
+      event.args.data.had_recent_input = false;
+    }
+
+    let traceHasWeightedScore = true;
+    const cumulativeShift = layoutShiftEvents
       .map(e => {
-        // @ts-expect-error Events without score are filtered out.
-        return /** @type {number} */ (e.args.data.score);
+        if (e.args.data.had_recent_input) return 0;
+
+        // FIXME: remove after 89 hits stable
```

Comment:
@paulirish likes `COMPAT`

Reply:
yeah I guess this is more COMPAT than FIXME, sorry @adamraine